src/index.js

```diff
@@ -316,6 +316,10 @@ const expressions = Object.assign(Object.create(null), {
    */
 
   $exists(expectedPresence, value, key, object) {
+    if (object === undefined) {
+      const isPresent = value !== undefined;
+      return isPresent === expectedPresence;
+    }
```

Comment:
I find difficulty following the logic
We're looking for non-undefined values and asserting existence against the expected?
This is to cover the case where the `object` is `undefined` and the `value` is `undefined` to return `true`?

Reply:
@AhmedFat7y 
To get more context please check [this example](https://github.com/mixmaxhq/sift.js#exists)

I haven't checked implementation in deep, but in case if `$exists` is used to just check primitive value, like with `$gt`, `$ne` or other operators, it has only `expectedPresence` and `value` as arguments, others are undefined and `has` function fails in this case, but it should not, instead it should just check if the value is undefined or not according to the `$exists: <is_not_undefined>`